### PR TITLE
feat: fetch and save balances histories

### PR DESCRIPTION
**Pull request by drazik**
We begin to store balance history. For that, we introduced a new doctype : io.cozy.bank.balancehistories. Each document of this doctype stores a year of daily balance.

You can see the documentation of this doctype here : https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.bank.md#iocozybankbalancehistories

It would be great if we can ask @carrieje to test a specifically built version of the connector on a testing instance before merging this PR because since I don't have an account in this bank, I developed it totally blindly (just a sort a copy/paste of what we did in the Linxo connector). What do you think ?

### CODE REVIEW

**Comment by drazik** [comment]
Anybody to review this one please ?

File: src/index.js [hunk 1]
```diff
@@ -190,3 +194,66 @@ function parseAmount(amount) {
...
   return moment.tz(date, 'D MMM YYYY', 'Europe/Paris').format()
 }
+
+async function getBalanceHistory(year, accountId) {
+  const index = await cozyClient.data.defineIndex(
+    'io.cozy.bank.balancehistories',
+    ['year', 'relationships.account.data._id']
+  )
+  const options = {
+    selector: { year, 'relationships.account.data._id': accountId },
```
Comment (ptbrowne) [inline src/index.js]: Is it possible with `relationships` to only have `relationship.account._id` ?

File: src/index.js [hunk 2]
```diff
@@ -190,3 +194,66 @@ function parseAmount(amount) {
...
+    year,
+    balances: {},
+    metadata: {
+      version: 1
+    },
+    relationships: {
+      account: {
+        data: {
+          _id: accountId,
+          _type: 'io.cozy.bank.accounts'
```
Comment (ptbrowne) [inline src/index.js]: Is the `_type` stored inside the db ? I found it weird to have it in the `data` if it's not stored in the db.

File: src/index.js [hunk 3]
```diff
@@ -190,3 +194,66 @@ function parseAmount(amount) {
...
+      history.balances[todayAsString] = account.balance
+
+      return history
+    })
+  )
+}
+
+function saveBalances(balances) {
+  return updateOrCreate(balances, 'io.cozy.bank.balancehistories', ['_id'])
+}
```
Comment (ptbrowne) [inline src/index.js]: All these functions working on balance histories could be put into the doctypes repository.

Comment (drazik) [reply src/index.js]: Yep :+1:

**Comment by drazik** [comment]
Closing as it's quite old and we finally don't use this doctype in banks.

**Pull request closed without merging**